File: tests/close-active-subaccounts/close-active-subaccounts.js
const AWS = require('aws-sdk');
const org = new AWS.Organizations({region: 'us-east-1'});
const retry = require('async-retry')
const lib = require('./lib')

const CAPTCHA_KEY = process.env['CAPTCHA_KEY'];

(async () => {
    try {
        const accounts = await org.listAccounts().promise();

        for (let account of accounts.Accounts) {
            if (account.JoinedMethod !== 'CREATED' || account.Status !== 'ACTIVE') {
                continue;
            }

            await retry(async bail => {
                console.log(account);
                await lib.deleteAccount(account.Email, CAPTCHA_KEY);
            }, {
                retries: 10
            });
        }
    } catch (e) {
        console.log('got exception in outer scope', e)
        process.exit(1);
    }
})();

File: tests/close-active-subaccounts/close-active-subaccounts.py
import boto3
import botocore
org = boto3.client('organizations', region_name='us-east-1')


def organizations_close_account(account):
    print(account)
    try:
        response = org.close_account(
            AccountId=account['Id']
        )
        print(response)
    except org.exceptions.AccountAlreadyClosedException:
        print("Account already closed")
    except botocore.exceptions.ClientError as e:
        print(e)


try:
    accounts = org.list_accounts()['Accounts']
except botocore.exceptions.ClientError as e:
    if e.response['Error']['Code'] == 'AWSOrganizationsNotInUseException':
        print("No AWS Organization found, nothing to do")
        exit(0)
    raise e

for account in accounts:
    if account['JoinedMethod'] != 'CREATED' or account['Status'] != 'ACTIVE':
        continue

    organizations_close_account(account)
